lenses_python/lenses.py

```diff
@@ -103,291 +183,95 @@ def SqlHandler(self, query, extract_pandas=0, datetimelist=[], formatinglist=[])
         return SqlH(self.url, self.username, self.password, self.token, query, datetimelist, formatinglist).ExecuteSqlQuery(extract_pandas)
 
     # Topics Handler
+    def ManageTopics(self, action, *opts, **kopts):
+        from lenses_python.TopicHandler import TopicHandler as TopicH
```

Comment:
> wait, what you are doing here is more like a CLI approach instead of a software API. if I want to see only the `TopicInfo` I need to pass a dict right now which is not that declarative. Abstractions are cool but when they are needed, right now you hide a couple of things under one method which later can be vulnerable and hard to debug because of different possible cases.
> 
> for example:
> 
> if I pass
> 
> ```
> .ManageTopics('lalalalalala')
> ```
> 
> which is an action which doesn't exist, your dict index will fail `action_dict[action]` as index doesn't exist (also you should raise an exception in this case because it's bad action). Apart from that what will happen if I won't pass any opts what the method will do?
> 
> This approach creates lots of questions and uncertainty of what the method need and does in practice. Your API should be declarative with concrete methods which will make a 3rd party person (a Data Engineer and not an expert) to understand what he need to do to fetch:
> 
>     * topic info
> 
>     * get all topics
>       etc.
> 
> 
> With the current approach needs a lot of guessing.

I will make the appropriate changes and include exception handlers based on your suggestions

Reply:
> Generally what we should optimise here is something different. We should think how we can avoid creating everytime an object of `TopicH` which in yours and previous practice still happen. I would prefer for example something like:
> 
> ```
> topic = Topic(......) ---> this will be created once and reuse it
> topic. GetAllTopics()
> ```
> 
> instead of what we do right now:
> 
> ```
> lenses.GetAllTopics() -> we create a new object
> lneses.TopicInfo() -> we create another new same object for the same topic
> ```
> 
> ^ the last one creates couple of objects for the same thing

Okay, I'll alter the code in order to create always one object and reuse it.